addons/web_editor/static/src/components/upload_progress_toast/upload_service.js

```diff
@@ -166,7 +166,181 @@ export const uploadService = {
                         throw error;
                     }
                 }
-            }
+            },
+            /**
+             * TODO: merge with uploadFiles in master
+             * This version removes service-level state and returns { promise, abort }.
+             */
+            uploadFilesWithAbort(
```

Comment:
No need to talk about the difference between both POC PRs in the actually merged comment ^_^
Maybe say that it is the same signature as uploadFiles and describe the returned value.

It is a bit sad that this is so difficult to compare with the other method while there are so few changes.
Maybe describe the actual differences in the comment.
From what I see:
- return "invalid_size" instead of null when a file is too big
- detect abort signal between uploads and at the end
- check for `lengthComputable` during progress